stable_baselines3/common/vec_env/dummy_vec_env.py

```diff
@@ -73,19 +75,20 @@ def step_wait(self) -> VecEnvStepReturn:
 
     def seed(self, seed: Optional[int] = None) -> Sequence[Union[None, int]]:
         # Avoid circular import
-        from stable_baselines3.common.utils import compat_gym_seed
-
+        # from stable_baselines3.common.utils import compat_gym_seed
         if seed is None:
-            seed = np.random.randint(0, 2**32 - 1)
-        seeds = []
-        for idx, env in enumerate(self.envs):
-            seeds.append(compat_gym_seed(env, seed=seed + idx))  # type: ignore[func-returns-value]
-        return seeds
+            self.seed = [None for _ in range(len(self.envs))]  # seed to be used in the next env.reset()
+            return
+        
+        self.seed = [seed + i for i in range(len(self.envs))]
+        return self.seed
 
     def reset(self) -> VecEnvObs:
         for env_idx in range(self.num_envs):
-            obs, self.reset_infos[env_idx] = self.envs[env_idx].reset()
+            obs, self.reset_infos[env_idx] = self.envs[env_idx].reset(seed=self.seed)
             self._save_obs(env_idx, obs)
+
+        self.seed = [None for _ in range(len(self.envs))]
```

Comment:
I would add a comment here, and we can probably move a `_reset_seeds()` method to the base class